src/lib.rs

```diff
@@ -1425,6 +1477,11 @@ impl Witness {
         ))
     }
 
+    // Witness for a utxo-based transaction generated externally (such as hardware wallets)
+    pub fn for_external_utxo(witness: &UtxoWitness) -> Witness {
```

Comment:
I'm not sure I get the meaning of this names (this and `for_external_account`). Isn't this more like a conversion (`from_...`)? I think it should be different from `for_account` and `for_utxo`, as those two actually perform the signing and create the witness (although I'm not saying those two are better named neither).

Reply:
yes you're right. Changed to this 👍